Restructuring readme (#219)
fixie-ai/ai-jsx: readme.md

[![Discord Follow](https://dcbadge.vercel.app/api/server/MsKAeKF8kU?style=flat)](https://discord.gg/MsKAeKF8kU)
[![Twitter Follow](https://img.shields.io/twitter/follow/fixieai?style=social)](https://twitter.com/fixieai)

## About AI.JSX

> AI.JSX is a framework for building AI applications using Javascript and JSX. With AI.JSX, you can easily orchestrate multiple Large Language Model (LLM) calls expressed as modular, reusable components, and interweave UI components with your AI components. This means you can rely on the LLM to construct your UI dynamically from a set of provided React components. AI.JSX also provides native support for tools, Document Question & Answering, and much more.

Reviewer: I actually think we we can simplify this paragraph even more.

Instead of "orchestrate multiple LLM calls expressed as..." let's just call it what it is: great tools for prompt engineering. Instead of "interweave UI components with AI components", let's just say you can have the LLM render React components in its response instead of just text.

Basically, I think we should evolve this to the eli5 version

Reviewer 2: 💯 for the "above the fold README" being ELI5.

Author: just committed an update to simplify per Zach's feedback...thoughts?

AI.JSX can be used to create standalone LLM applications that can be deployed anywhere Node.js is supported, or it can be used as part of a larger React application.

## Features

AI.JSX comes with the following features out-of-the-box:

- **Componetized** → LLM prompt engineering through modular, reusable components.

> - **Model Support** → Use Llama2 and models from Anthropic and OpenAI. Seamlessly switch between model providers and LLM config (e.g. temperature).

Reviewer 2: ```suggestion
- **Model Support** → Use OpenAI, Antrhopic, and Llama2, or bring your own model. Seamlessly switch between model providers and LLM config (e.g. temperature).
```

> - **Complete AI Toolbox** → Built-in support for Tools (ReAct pattern), Document Question and Answering, Chain of Thought, and more.

Reviewer 2: I don't think we actually use the `ReAct` pattern.

Reviewer 2: Or CoT

Author: removing these

> - **Generative UI** → Seamlessly interweave LLM calls with standard UI components. LLM can dynamically render UI from a set of provided components.

Reviewer 2: ```suggestion
- **Generative UI** → Seamlessly interweave LLM calls with standard UI components. LLM can dynamically render UI from a set of components you provide.
```
